### DISCUSSION TITLE
Blocking render call?

### DISCUSSION THREAD

**Discussion by yavuz650** [google/filament#6563] [Q&A] [answered]
Apparently `filament::Renderer::render()` function is nonblocking, so when I call `renderer->render(view)` in my loop, the CPU does not wait for the rendering process to finish, much like GL draw calls.

Is there a way to force the CPU to wait until Filament finishes rendering the frame, kinda like `glFinish()`?

**Comment by romainguy** [comment, marked as answer]
You can use `Engine::flushAndWait()` but it's not recommended under normal operation (since you will indeed block the current thread). To drive the render loop you should instead use vsync signals and the value returned by `Renderer::beginFrame()`.

**Reply by yavuz650** [reply]
I'm doing this for profiling purposes, so it's not really normal operation. I'm trying to capture the draw calls and see what the GPU utilization is like. Thank you for the hint!
